app/src/main/java/com/example/househomey/UserProfileFragment.java

```diff
@@ -0,0 +1,61 @@
+package com.example.househomey;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.Looper;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+
+import com.google.firebase.auth.FirebaseAuth;
+
+public class UserProfileFragment extends Fragment {
+
+    private String username;
+    TextView usernameTextView;
+    Button logoutButton;
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+
+        // Inflate the fragment's layout
+        View rootView = inflater.inflate(R.layout.fragment_user_profile, container, false);
+
+        //Populate the item list from bundle
+        Bundle args = getArguments();
+
+        this.username = args.getString("username");
```

Comment:
do we need to bundle this? Should be able to use firebase auth to get the display name

Reply:
I changed it but had to add a check for if the user is signed out somehow.